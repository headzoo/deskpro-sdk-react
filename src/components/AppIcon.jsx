import React from 'react';
import PropTypes from 'prop-types';
import Badge from './Badge';

const AppIcon = ({ badgeVisible, badgeCount, style, ...props }) => {
  const styles = Object.assign({}, {
    position:    'relative',
    marginTop:   6,
    marginRight: 6
  }, style);

  return (
    <div {...props} style={styles}>
      <Badge
        count={badgeCount}
        visible={badgeVisible}
      />
      <img
        src="../assets/icon.png"
        style={{ width: 16, height: 16, border: 0 }}
      />
    </div>
  );
};

AppIcon.propTypes = {
  /**
   * Whether or not the badge count is displayed.
   */
  badgeVisible: PropTypes.bool,
  /**
   * The badge count.
   */
  badgeCount:   PropTypes.number,
  /**
   * Additional styles passed to the room element.
   */
  style:        PropTypes.object
};

AppIcon.defaultProps = {
  badgeVisible: false,
  badgeCount:   0,
  style:        {}
};

export default AppIcon;
