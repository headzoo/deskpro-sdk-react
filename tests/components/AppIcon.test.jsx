import React from 'react';
import renderer from 'react-test-renderer';
import AppIcon from '../../src/components/AppIcon';

it('AppIcon renders a badge', () => {
  const renderedValue = renderer.create(
    <AppIcon badgeCount={3} badgeVisible />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});

it('AppIcon hides the badge', () => {
  const renderedValue = renderer.create(
    <AppIcon badgeCount={3} badgeVisible={false} />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
