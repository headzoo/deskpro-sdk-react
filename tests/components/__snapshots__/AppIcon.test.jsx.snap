// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon hides the badge 1`] = `
<div
  style={
    Object {
      "marginRight": 6,
      "marginTop": 6,
      "position": "relative",
    }
  }
>
  <img
    src="../assets/icon.png"
    style={
      Object {
        "border": 0,
        "height": 16,
        "width": 16,
      }
    }
  />
</div>
`;

exports[`AppIcon renders a badge 1`] = `
<div
  style={
    Object {
      "marginRight": 6,
      "marginTop": 6,
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "position": "absolute",
        "right": -10,
        "top": -13,
      }
    }
  >
    <div
      className="dp-badge dp-badge--danger dp-bg--danger dp-circle-badge"
    >
      3
    </div>
  </div>
  <img
    src="../assets/icon.png"
    style={
      Object {
        "border": 0,
        "height": 16,
        "width": 16,
      }
    }
  />
</div>
`;
